lib/utils.js

```diff
@@ -107,11 +107,16 @@ var assign = function assignSingleSource(target, source) {
     }, target);
 };
 
-var decode = function (str) {
+var decode = function (str, decoder, charset) {
+    var strWithoutPlus = str.replace(/\+/g, ' ');
+    if (charset === 'iso-8859-1') {
+        return unescape(strWithoutPlus); // Cannot throw
```

Comment:
I think `unescape` will decode the format %uXXXX in addition to the standard %XX format. This seems like an undeserved behavior, and at best just weird that the %uXXXX format will work if you use latain 1 but not if you use utf8. It should probably either accept %uXXXX in both or neither.

Reply:
`unescape` is a legacy global function that converts string => string, interpreting percent-encoded octets to be in iso-8859-1. Percent not followed by two hex chars are left as-is, and I have never succeeded in getting it to throw for any string value, so that's why it doesn't seem necessary to put it in a try...catch.

`decodeURIComponent` is similar, but interprets the percent-encoded octets as utf-8. It'll throw a `URIError` if percent is not followed by two hex chars, or if sequences of percent-encoded octets don't form a valid utf-8 character.

Reply:
> Percent not followed by two hex chars are left as-is

This is not correct,  see my comment above yours about %uXXXX . Can you make it not decode that format and only touch the %XX ?

Reply:
Ah, you're right. I had forgotten about that. Fixed in bd9bc06d2bff26d1a9c83662e3453d0770da0f23.